mungegithub/mungers/issue-creator.go

```diff
@@ -0,0 +1,396 @@
+	return ((*github.Config)(config)).NewIssue(title, body, labels, owners)
+}
+
+//Issue is an interface implemented by structs that can be synced with github issues via the IssueCreator.
+type Issue interface {
+	//Title yields the initial title text of the github issue.
+	Title() string
+	//Body yields the body text of the github issue and *must* contain the output of ID().
+	//closedIssues is a (potentially empty) slice containing all closed
+	//issues authored by this bot that contain ID() in their body.
+	//if Body returns an empty string no issue is created.
+	Body(closedIssues []*github.MungeObject) string
+	//ID returns a string that uniquely identifies this issue.
+	//This ID must appear in the body of the issue.
+	//DO NOT CHANGE how this ID is formatted or duplicate issues will be created
+	//on github for this issue
+	ID() string
+	//Labels specifies the set of labels to apply to this issue on github.
+	Labels() []string
+	//Owners returns the github usernames to assign the issue to or nil/empty for no assignment.
+	Owners() []string
+	//Priority calculates and returns the priority of this issue
+	//The returned bool indicates if the returned priority is valid and can be used
+	Priority() (int, bool)
+}
+
+//IssueCreator handles syncing identified issues with github issues.
+//This includes finding existing github issues, creating new ones, and ensuring that duplicate
+//github issues are not created.
+type IssueCreator struct {
+	//config is the github client that is used to interact with github.
+	config RepoClient
+	//validLabels is the set of labels that are valid for the current repo (populated from github).
+	validLabels []*githubapi.Label
+	//authorName is the name of the current bot.
+	authorName string
+	//allIssues is a local cache of all issues in the repo authored by the currently authenticated user.
+	//MungeObjects are keyed by issue number
+	allIssues map[int]*github.MungeObject
+
+	//ownerPath is the path the the test owners csv file or "" if no assignments or SIG areas should be used.
+	ownerPath string
+	//maxSIGCount is the maximum number of SIG areas to include on a single github issue.
+	maxSIGCount int
+	//maxAssignees is the maximum number of user to assign to a single github issue.
+	maxAssignees int
+
+	//owners is an OwnerMapper that maps test names to owners and SIG areas.
+	owners OwnerMapper
+}
+
+func init() {
+	RegisterMungerOrDie(&IssueCreator{})
+}
+
+//Initialize prepares an IssueCreator for use by other mungers.
+//This includes determining the currently authenticated user, fetching all issues created by that user
+//from github, fetching the labels that are valid for the repo, and initializing the test owner and sig data.
+func (creator *IssueCreator) Initialize(config *github.Config, feats *features.Features) error {
+	cfg := githubConfig(*config)
+	creator.config = RepoClient(&cfg)
+
+	//load test owner/SIG data
+	var err error
```

Comment:
Ack. (Cannot define a new variable and assign to a field in a single assignment statement in go)